pilot/pkg/model/push_context.go

```diff
@@ -135,7 +135,39 @@ type XDSUpdater interface {
 	// ConfigUpdate is called to notify the XDS server of config updates and request a push.
 	// The requests may be collapsed and throttled.
 	// This replaces the 'cache invalidation' model.
-	ConfigUpdate(full bool)
+	ConfigUpdate(req UpdateReq)
+}
+
+// UpdateReq defines a request to update proxies
+type UpdateReq struct {
+	// Full determines whether a full push is required or not. If set to false, only endpoints will be sent.
+	Full bool
+
+	// UpdateNamespaces contains a list of namespaces that were changed in the update.
+	// This is used as an optimization to avoid unnecessary pushes to proxies that are scoped with a Sidecar.
+	// Currently, this will only scope EDS updates, as config updates are more complicated.
+	// If this is empty, then proxies in all namespaces will get an update
+	// If this is present, then only proxies that import this namespace will get an update
+	UpdateNamespaces map[string]struct{}
```

Comment:
nit: after reading the PR, I think this reads better as TargetNamespaces.

Reply:
:+1: I didn't like this name either, I am bad at naming